src/Compilers/Core/Portable/Symbols/Attributes/CommonAttributeData.cs

```diff
@@ -74,13 +80,55 @@ internal static bool IsTargetEarlyAttribute(INamedTypeSymbol attributeType, int
                 // NOTE: Below assumption disallows early decoding well-known attributes with optional parameters.
                 if (attributeArgCount == parameterCount)
                 {
-                    string actualNamespaceName = attributeType.ContainingNamespace.ToDisplayString(SymbolDisplayFormat.QualifiedNameOnlyFormat);
                     StringComparison options = description.MatchIgnoringCase ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal;
-                    return actualNamespaceName.Equals(description.Namespace, options) && attributeType.Name.Equals(description.Name, options);
+                    return attributeType.Name.Equals(description.Name, options) && namespaceMatch(attributeType.ContainingNamespace, description.Namespace, options);
                 }
             }
 
             return false;
+
+            static bool namespaceMatch(INamespaceSymbolInternal container, string namespaceName, StringComparison options)
```

Comment:
This actually also addresses [https://github.com/dotnet/roslyn/issues/37345](https://github.com/dotnet/roslyn/issues/37345).

Reply:
> This actually also addresses [https://github.com/dotnet/roslyn/issues/37345](https://github.com/dotnet/roslyn/issues/37345).

This eliminates one use of the API, but doesn't change how expensive the API is.

---
In reply to: [339150937](https://github.com/dotnet/roslyn/pull/39498#discussion_r339150937) [](ancestors = 339150937)